pallets/ethereum-transactions/src/lib.rs

```diff
@@ -712,3 +677,39 @@ impl<BlockNumber: Member + AtLeast32Bit> DispatchedData<BlockNumber> {
         return DispatchedData::<BlockNumber> { transaction_id, submitted_at_block }
     }
 }
+
+// A value placed in storage that represents the current version of the EthereumEvents pallet
+// storage. This value is used by the `on_runtime_upgrade` logic to determine whether we run its
+// storage migration logic.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, MaxEncodedLen, TypeInfo)]
+enum Releases {
+    Unknown,
+    V2_0_0,
+    V3_0_0,
+    V4_0_0,
+}
+
+impl Default for Releases {
+    fn default() -> Self {
+        Releases::V4_0_0
+    }
+}
+
+pub mod migrations {
+    use super::*;
+
+    pub fn migrate_pb_to_bridge_contract<T: Config>() -> frame_support::weights::Weight {
+        sp_runtime::runtime_logger::RuntimeLogger::init();
+        log::info!("ℹ️  Ethereum Transactions pallet data migration invoked");
+
+        let mut consumed_weight = T::DbWeight::get().reads_writes(0, 1);
+
+        log::info!("Destroying contract........");
+        <PublishRootContract<T>>::kill();
```

Comment:
I realise we probably have added this contract from the other pallet, but just in case we haven't shouldn't we check if the root contract is in the new structure before we kill it?

Reply:
Probably a nitpick :/